lib/routes.js

```diff
@@ -21,58 +55,64 @@ var setup = function (app, router) {
   router.get('/debug', (req, res) => {
     logger.info("🛎️  Route /debug")
     logger.level = 'debug'
-    res.status(200).json({ "value": "Debug Mode On"})
-  });
-
-  router.get('/stop', (req, res) => {
-    logger.info("🛎️  Route /stop")
-    res.status(200).json({ "value": "Stopping application"})
-    process.exit();
+    res.status(200).json({ "value": "Debug Mode On" })
   });
 
   router.get('/read/:ip/:port/:unitId/:regType/:startAddress/:quantity', (req, res) => {
-    // Example
-    // http://localhost:20900/read/192.168.255.1/502/1/holding/4110/1/float?offset=111&multiplier=2
-    logger.info("🛎️  Route /read/... " + JSON.stringify(req.params))
-
+    logger.info("🛎️  Route /read Queued")
     logger.debug("🧩 Incoming Params: " + JSON.stringify(req.params));
     logger.debug("🔑 Incoming keys: " + JSON.stringify(req.query));
 
-    let reading = null
-
-    // Conversion Values
-    req.query.conversion = (req.query.hasOwnProperty('conversion')) ? req.query.conversion : "raw"
-    req.query.endianness = (req.query.hasOwnProperty('endianness')) ? req.query.endianness : "little"
-    req.query.debug = (req.query.hasOwnProperty('debug')) ? req.query.debug : false
-
-
-    // Reading
-    modbus.readRegister(
-      req.params.ip,
-      req.params.port,
-      req.params.unitId,
-      req.params.regType,
-      req.params.startAddress,
-      req.params.quantity,
-      req.query.conversion,
-      req.query.endianness
-    ).then((reading) => {
-      if (isNaN(reading) || reading == null) {
-        res.status(400).json({ "params": req.params, "reading": String(reading), "error": "NaN" })
-      } else {
-        // Scaling
-        req.query.multiplier = (req.query.hasOwnProperty('multiplier')) ? req.query.multiplier : 1
-        req.query.offset = (req.query.hasOwnProperty('offset')) ? req.query.offset : 0
-        const transformedReading = transformReading(reading, req.query.offset, req.query.multiplier)
-
-        // Respond
-        if (req.query.debug) {
-          res.status(200).json({ "params": req.params, "query": req.query, "reading": reading, "transformedReading": transformedReading })
+    // Add item to queue with callback
+    q.push({ name: 'Modbus Reading' }, function (err) {
```

Comment:
It's important here that this function pushed onto the queue is also `async` and `awaits` everything. As written you are using `.then` instead of `await`, which causes this callback to return immediately.

The `async` documentation is not very clear here, but this is not the correct call. This form of call to `q.push` pushes the value `{ name: "Modbus Reading" }` onto the queue, so this is what will come out in `callback` in your queue-processing function above. This will fail when the code above tries to call `callback()`. The function passed as second argument is used for custom error-handling for this queue item.

Instead, you want `q.push(async function () {...})`. This pushes an async function onto the queue (the function itself is a data item), which you can then call (and await) when you process the queue.